packages/labs/react/src/create-component.ts

```diff
@@ -257,7 +236,12 @@ export const createComponent = <I extends HTMLElement, E extends Events>(
       for (const [k, v] of Object.entries(this.props)) {
         if (k === '__forwardedRef') continue;
 
-        if (elementClassProps.has(k)) {
+        if (
+          eventProps.has(k) ||
+          (!reservedReactProperties.has(k) &&
+            !(k in HTMLElement.prototype) &&
+            k in elementClass.prototype)
```

Comment:
We still need to verify this logic (specifically excluding `k in HTMLElement.prototype`) with @josepharhar to see if it matches React 19. I'm good with going in as-is now though.